src/RequestHandle.cpp

```diff
@@ -389,6 +443,21 @@ auto RequestHandle::setCompletionStatus(
         case CURLcode::CURLE_SSL_CONNECT_ERROR:
             m_status_code = RequestStatus::CONNECT_SSL_ERROR;
             break;
+        case CURLcode::CURLE_WRITE_ERROR:
+            /**
+             * If there is a cURL write error, but the maximum number of bytes
+             * has been written, then we intentionally aborted, so let's set
+             * this to success. Otherwise, it's a legitimate error.
+             */
+            if (getBytesLeftToWrite() == 0)
```

Comment:
I would suggest this feels like a good use of a ternary since it is only assigning m_status_code

m_status_code = (getRemainingDownloadBytes() == 0)
    ? RequestStatus::SUCCESS
    : RequestStatus::ERROR

Reply:
Updated